File: Main/Models/SharedCollection.cs
﻿
using Newtonsoft.Json;

namespace LinkCollectionApp.Models
{
  public partial class SharedCollection
  {
    public int CollectionId { get; set; }
    public string UserId { get; set; }
    public bool EditRights { get; set; }

    public virtual Collection Collection { get; set; }
    [JsonIgnore]
    public virtual ApplicationUser User { get; set; }
  }
}

File: Main/ClientApp/src/Model/SharedCollection.ts
import { Collection } from "./Collection";
import { User } from "./User";

export type SharedCollection = {
  collectionId: number;
  collection: Collection;
  userId: string;
  editRights: boolean;
  user: User;
};

export type SharedCollectionData = {
  collectionId: number;
  userId: string;
  editRights: boolean;
};
